src/actions/VaccineActions.js

```diff
@@ -169,33 +193,43 @@ const startDownloadAllLogs = macAddress => async (dispatch, getState) => {
   return null;
 };
 
-const setLogInterval = (macAddress, interval) => async dispatch => {
-  let ok = false;
-  let error = `Sensor response was not equal to 'Interval: ${interval}s'`;
-
+const setLogInterval = (macAddress, interval) => dispatch => {
   dispatch(setLogIntervalStart(macAddress));
+  const regex = new RegExp(`Interval: ${interval}s`); // TODO: update with sensor specific response as needed
+  const error = `Sensor response was not equal to 'Interval: ${interval}s'`;
+
+  return BleService()
+    .updateLogInterval(macAddress, interval)
+    .then(result =>
+      regex.test(result.toString()) ? setLogIntervalSuccess() : setLogIntervalError(error)
+    )
+    .then(action => dispatch(action));
+};
 
-  try {
-    const result = await BleService().updateLogIntervalWithRetries(macAddress, interval, 3, error);
-    const regex = new RegExp(`Interval: ${interval}s`); // TODO: update with sensor specific response as needed
-    ok = regex.test(result);
-  } catch (e) {
-    error = e;
-    ok = false;
-  }
-
-  if (ok) {
-    dispatch(setLogIntervalSuccess());
-    ToastAndroid.show(
-      vaccineStrings.formatString(vaccineStrings.log_interval_set, interval),
-      ToastAndroid.LONG
-    );
-  } else {
-    ToastAndroid.show(vaccineStrings.E_COMMAND_FAILED, ToastAndroid.LONG);
-    dispatch(setLogIntervalError(error));
-  }
-
-  return ok;
+const saveSensor = sensor => dispatch =>
+  new Promise((resolve, reject) => {
+    dispatch(saveSensorStart(sensor.macAddress));
+    try {
+      const { location, logInterval, macAddress, name } = sensor;
+      const sensorManager = SensorManager();
+      const newSensor = sensorManager.createSensor({ location, logInterval, macAddress, name });
+      sensorManager.saveSensor(newSensor);
+      dispatch(saveSensorSuccess());
+      resolve();
+    } catch (error) {
+      dispatch(saveSensorError());
+      reject(error);
+    }
+  });
+
+const disableSensorButton = macAddress => dispatch => {
```

Comment:
```
const disableSensorButton = macAddress => async dispatch => {
  dispatch(disableButtonStart(macAddress));

  try {
    const info = BleService().getInfoWithRetries(macAddress, 10);
    if (info.isDisabled) {
      await BleService().toggleButtonWithRetries(macAddress, 10);
    }
  } catch {
    // We're not doing anything in this case?
  } finally {
    dispatch(disableButtonStop(macAddress));
  }
};
```

Reply:
the error was being raised to the caller, so the empty `catch` here would suppress it